spec/views/user_show_spec.rb

```diff
@@ -43,19 +22,28 @@
     end
   end
 
-  it 'displays the user name and profile image' do
-    expect(page).to have_content(@users.first.name)
-    expect(page).to have_content(@users.first.bio)
-    expect(page.has_xpath?("//img[@src='#{@users.first.photo}']")).to be true
-    expect(page).to have_content("Number of posts: #{@users.first.posts_counter}")
+  it 'I can see the number of posts the user has written.' do
+    visit '/user'
+    page.has_content?(user.posts_counter)
+  end
+
+  it "I can see the user's bio." do
+    visit '/user'
+    page.has_content?(user.bio)
   end
 
-  it 'displays a section for pagination' do
-    Post.create(title: 'Post 6', text: 'First Time Home Buyer Tips', comments_counter: 2, likes_counter: 3)
-    Post.create(title: 'Post 7', text: 'Job interview tips', comments_counter: 2, likes_counter: 5)
-    Post.create(title: 'Post 8', text: 'Nature Photography', comments_counter: 3, likes_counter: 1)
-    Post.create(title: 'Post 9', text: 'Preparing for a marathon', comments_counter: 1, likes_counter: 2)
-    Post.create(title: 'Post 10', text: 'Favorite cooking recipes', comments_counter: 0, likes_counter: 7)
+  before(:each) do
     visit '/user'
   end
+
+  it "I can see a button that lets me view all of a user's posts." do
+    visit 'user'
+    page.has_button?('See all posts')
+  end
+
+  it "When you go see all posts, it redirects me to the user's post's index page." do
+    visit 'user'
+    visit 'user/1post'
+    page.has_content?(user.name)
+  end
```

Comment:
- [ ] Let's ensure to include test for the below scenarios for _user show_ views: 
   * I can see the user's profile picture
   * I can see the user's username
   * I can see the user's first 3 posts
   * When I click a user's post, it redirects me to that post's show page